### ISSUE TITLE
[Master feature] AMP Google Pay

### ISSUE THREAD

**Issue by choumx** [ampproject/amphtml#15201]
Adds support for integrating Google Pay experiences into AMP pages.

## Motivation

We want to make payments in AMP pages easy and fast by using [Google Pay](https://pay.google.com/about/). We plan to add two new AMP components: `amp-payment-google-inline` and `amp-payment-google-button`. 

The goal is to eventually make these extensions work on both Google and non-Google surfaces.

## Requirements

- Support Google Pay with two UI options: an inline widget and a button that displays a popover.

## Proposed design

- `amp-payment-google-inline`: Renders an inline widget that preselects one of the user's payment methods.
- `amp-payment-google-button`: Renders a button that triggers a popover to slide in where the user can select their desired payment method and (optional) shipping address. 

### `amp-payment-google-inline`

To use the `amp-payment-google-inline` component, the AMP developer adds a `<amp-payment-google-inline>` tag with a child `<script>` element that contains a [PaymentDataRequest](https://developers.google.com/pay/api/web/object-reference#PaymentDataRequest) JSON object. 

This tag should also have a `data-payment-data-input-id` attribute which references the `id` of a hidden input field in a form on the page that will be filled with the [PaymentData](https://developers.google.com/pay/api/web/object-reference#PaymentData) JSON object when the user submits the form. 

The `amp-payment-google-inline` tag also fires a `paymentsReadyStatusChanged` event, so you can know when the user has a payment ready and the form can be submitted.

```html
<amp-payment-google-inline
    data-payment-data-input-id="payment-data"
    on="paymentReadyStatusChanged: AMP.setState({isPaymentReady: event})">
    <script type="application/json">
        {
            "merchantId": "your merchant ID",
            ...
        }
    </script>
</amp-payment-google-inline>
<input type="hidden" id="payment-data" name="payment-data">
```

### `amp-payment-google-button`

Similarly, to use the amp-payment-google-button component, the AMP developer adds a `<amp-payment-google-button>` tag with a child `<script>` element that contains a [PaymentDataRequest](https://developers.google.com/pay/api/web/object-reference#PaymentDataRequest) JSON object. 

Rather than filling a hidden input field, this tag will simply fire a `paymentReadyStatusChanged` event when the user has clicked the button and chosen a payment method. The payload of this event will be a [PaymentData](https://developers.google.com/pay/api/web/object-reference#PaymentData) JSON object.

```html
<amp-payment-google-button
    data-payment-data-input-id="payment-data"
    on="loadPaymentData: AMP.setState({googlePaymentData: event})">
    <script type="application/json">
        {
            "merchantId": "your merchant ID",
            ...
        }
    </script>
</amp-payment-google-button>
```

**Comment by ampprojectbot** [comment]
This issue seems to be in  Pending Triage for awhile.  Please triage this to an appropriate milestone.

**Comment by cvializ** [comment]
`amp-form` is capable of automatically interacting and extracting values from other amp-elements using the new form submit service. So it might not be necessary for document authors to create their own hidden inputs or specify a `data-payment-data-input-id`.

**Comment by ampprojectbot** [comment]
This issue doesn't have a category which makes it harder for us to keep track of it.  Please add an appropriate category.

**Comment by morsssss** [comment]
I was just curious, as we're working on a sample e-comm demo that will eventually have a payment step... has this task been prioritized? Do we expect anyone to pick it up within the next few months?

Thanks!

**Comment by torch2424** [comment]
@morsssss We had this brought up during our design review: https://github.com/ampproject/amphtml/issues/18658

Seems like the Google Pay team is working on this, and planning on contributing it, so you may want to reach out to them? Also, their work is dependent on some of my work with Async Input, which is in it's final PR now before being code complete 😄

**Comment by ampprojectbot** [comment]
This issue doesn't have a category which makes it harder for us to keep track of it.  Please add an appropriate category.

**Comment by KungFuCookie** [comment]
Is there an indication when this will be available?

**Comment by torch2424** [comment]
@KungFuCookie As of right now from what I know, we don't have a set timeline for when this will be released. But it is definitely something we are working on.

I'll go ahead and reach out to the Google Pay team to see how are things on their end to see if I can get an answer for you 😄

**Comment by choumx** [comment]
This project has been canceled.
